spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoExceptionTranslator.java

```diff
@@ -78,21 +88,25 @@ public DataAccessException translateExceptionIfPossible(RuntimeException ex) {
 			throw new InvalidDataAccessApiUsageException(ex.getMessage(), ex);
 		}
 
-		String exception = ClassUtils.getShortName(ClassUtils.getUserClass(ex.getClass()));
+		Class<? extends RuntimeException> exceptionClass = ex.getClass();
```

Comment:
Let's add a specific `instanceof` check instead of introducing a pattern matching for a single case.

Reply:
I'm confuse about your remark because I dont see how to apply:
- if I apply `instanceOf MongoSocketException` it will work for my use-case of course, 
but the question is how to not apply this for the rest (of the translation matrix entries) ?

Should we consider accepting children  for all entries ?


In my first suggest, I consider this points:
- we would like to keep a minor coupling with mongo exception hierarchy. 
- we suppose that MongoSocketException update frequency is less than their children.
- we would like to avoid listing individually each MongoSocketException children.
That's why I introduced "Class$children" a pattern matching.

Reply:
We already depend on `MongoException` so adding a subtype is fine. We don't have specific mappings for `MongoSocketException` so it doesn't clash with any existing mappings. Also, the error code for all `MongoSocketException`s is `-2` so its fine to follow the simplemost approach. Any further functionality isn't required right now and the easiest way is typically the one which works best.

Reply:
>the easiest way is typically the one which works best.
100% agree with that, so I'll upd with a specific `MongoSocketException` case.